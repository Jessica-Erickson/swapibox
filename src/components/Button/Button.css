.Button {
  font-family: 'News Cycle', sans-serif;
  height: 45px;
  width: 150px;
  border: 2px solid gold;
  background-color: transparent;
  color: gold;
  font-size: 1.25rem;
  border-radius: 15px;
  padding-bottom: 3px;
  letter-spacing: 1px;
  text-decoration: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Button:hover {
  color: white;
  border-color: white;
}

.favorites {
  height: 75px;
  min-width: 100px;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-bottom: 7px;
}

.favorites:hover {
  box-shadow: 0 0 15px gold, 0 0 15px white;
}

.favorites-badge {
  font-size: 1.75rem;
  font-weight: 700;
  margin: 0;
  line-height: 1;
}

.people:hover {
  box-shadow: 0 0 15px deepskyblue, 0 0 15px white;
}

.planets:hover {
  box-shadow: 0 0 15px springgreen, 0 0 15px white;
}

.vehicles:hover {
  box-shadow: 0 0 15px red, 0 0 15px white;
}

.active {
  background-color: gold;
  color: black;
  font-weight: 700;
}

.active:hover {
  color: black;
  border-color: black;
}

.active:focus {
  outline: none;
}