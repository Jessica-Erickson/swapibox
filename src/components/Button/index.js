import React from 'react'
import PropTypes from 'prop-types'
import './Button.css'

const Button = ({ label , favorites , handleNavClick, isActive }) => {
  if (favorites) {
    return (
      <button className={`Button ${label} ${ isActive ? 'active' : '' }`}
              value={ label }
              onClick={() => handleNavClick(label)}
      >
        Favorites
        <div className="favorites-badge">
          {favorites}
        </div>
      </button>
    )
  } else {
    return (
      <input className={`Button ${label} ${ isActive ? 'active' : '' }`}
             type="button"
             value={ label }
             onClick={() => handleNavClick(label)}
      />
    )
  }
}

Button.propTypes = {
  label: PropTypes.string.isRequired,
  favorites: PropTypes.number,
  handleNavClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired
}

export default Button;