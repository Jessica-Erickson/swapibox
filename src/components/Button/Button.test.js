import React from 'react'
import { shallow , mount } from 'enzyme'
import Button from './index'

describe('Button', () => {
  let wrapper;
  let label;
  let favorites;
  let currentDisplay;
  let handleNavClick;
  let isActive;

  beforeEach(() => {
    label = 'People'
    favorites = []
    handleNavClick = jest.fn()
    currentDisplay = 'default'
  });

  it('should match the snapshot', () => {
    wrapper = shallow(<Button label={label}
                              handleNavClick={handleNavClick}
                              currentDisplay={currentDisplay}
                              isActive={isActive}
                      />)
    expect(wrapper).toMatchSnapshot()
  });

  it('should render favorites button', () => {
    wrapper = shallow(<Button label='Favorites'
                              favorites={favorites}
                              handleNavClick={handleNavClick}
                              currentDisplay={currentDisplay}
                              isActive={isActive}
                      />)

    expect(wrapper).toMatchSnapshot()
  });

  it('should call handleNavClick when category button clicked', () => {
    wrapper = mount(<Button
                      label={label}
                      handleNavClick={handleNavClick}
                      currentDisplay={currentDisplay}
                      isActive={isActive}
                    />);

    wrapper.find('.People').simulate('click');

    expect(handleNavClick).toHaveBeenCalled();
  });

  it('should call handleNavClick when favorites button clicked', () => {
    label = 'Favorites';

    wrapper = mount(<Button
                      label={label}
                      favorites={favorites}
                      handleNavClick={handleNavClick}
                      currentDisplay={currentDisplay}
                      isActive={isActive}
                    />);

    wrapper.find('.Favorites').simulate('click');

    expect(handleNavClick).toHaveBeenCalled();
  });

  it('should add "active" class if isActive', () => {
    isActive = true
    wrapper = shallow(<Button label={label}
                      handleNavClick={handleNavClick}
                      currentDisplay={currentDisplay}
                      isActive={isActive}
                      />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should add "active" class if favorites button isActive', () => {
    isActive = true
    label='Favorites'
    wrapper = shallow(<Button label={label}
                      favorites={favorites}
                      handleNavClick={handleNavClick}
                      currentDisplay={currentDisplay}
                      isActive={isActive}
                      />)

    expect(wrapper).toMatchSnapshot()
  })
})
