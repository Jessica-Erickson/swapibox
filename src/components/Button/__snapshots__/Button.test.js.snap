// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<input
  className="Button People "
  onClick={[Function]}
  type="button"
  value="People"
/>
`;
